net/interfaces/interfaces.go

```diff
@@ -537,10 +550,7 @@ func anyInterestingIP(pfxs []netaddr.IPPrefix) bool {
 
 // isInterestingIP reports whether ip is an interesting IP that we
```

Comment:
does this function need to exist anymore then?

Reply:
I think so, in that it suppresses loopback addresses and this PR doesn't treat loopback differently. This PR only concerns link local 169.254.x.x.

Reply:
I just mean the function is so simple now, does it make sense to just inline it at the callers instead?

It's fine keeping it, though.